src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+      || this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.a + this.c <= this.b
+    ) {
+      throw new Error(
+        `sides ${this.a}, ${this.b} and ${this.c} can't form a triangle`,
```

Comment:
Here the same
![image](https://user-images.githubusercontent.com/57747481/210356875-5669b371-a47e-47cf-9c0b-750f074747a6.png)